### PR TITLE
💄 JS files lose all unused vars and imports

### PR DESCRIPTION
**Pull request by jdhoffa**
This PR is where things get a little spicy. I am simply removing every `var` and `import` that is unused!
There are a few categories here:
- The simplest is just tech debt, a `var` or function argument was defined (and probably used at some point) and later removed from he function body, but left in the body
- `vars` that are defined for their side-effects (you can just remove them and call the methods alone)
- unused package imports (self explanatory)

This PR affects the following plots: 
- pie_exploded
- tech_exposure
- tech_exposure_company
- techmix_sector
- timeline
- trajectory_alignment

I will post side-by-sides of these plots before and after to show they still render (and I have visually inspected them to ensure all interactivity still functions as expected)

@MonikaFu FYI

Relates to #96 
Towards #29

### CODE REVIEW

**Comment by jdhoffa** [comment]
# pie_exploded
## before
Screenshot 2024-12-02 at 17 03 55

## after
Screenshot 2024-12-02 at 17 07 22


# tech_exposure
## before
Screenshot 2024-12-02 at 17 04 16

## after
Screenshot 2024-12-02 at 17 07 40


# tech_exposure_company
## before
Screenshot 2024-12-02 at 17 04 39

## after
Screenshot 2024-12-02 at 17 07 58


# techmix_sector
## before
Screenshot 2024-12-02 at 17 05 06

## after
Screenshot 2024-12-02 at 17 08 23


# timeline
## before
Screenshot 2024-12-02 at 17 06 06

## after
Screenshot 2024-12-02 at 17 08 44


# trajectory_alignment
## before
Screenshot 2024-12-02 at 17 05 41

## after
Screenshot 2024-12-02 at 17 08 59

**Review by AlexAxthelm — APPROVED** [review]

File: src/js/techmix_sector.js [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import * as d3 from 'd3';
-import { union } from 'd3-array';
```
Comment (cjyetman) [inline src/js/techmix_sector.js:2]: `union` is being used on line 130/129 no?

Comment (jdhoffa) [reply src/js/techmix_sector.js:2]: Indeed, however that `union` call is a native method for the Set prototype. 
Check it:

Adding the following console logging to the js there:
``` js
let subgroups = Array.from(subgroups0.union(subgroups1));
console.log(subgroups0.constructor.name);
console.log(Set.prototype.union);
```

and inspecting in the console, I see:
Screenshot 2024-12-02 at 19 19 14

Comment (jdhoffa) [reply src/js/techmix_sector.js:2]: And note that if I add the line back:
``` js
import { union } from 'd3-array';
```

and do the same logging, it yields the same source:
Screenshot 2024-12-02 at 19 20 12

We were never calling the `union` method from `d3-array` in the first place

Comment (jdhoffa) [reply src/js/techmix_sector.js:2]: Seems like its new: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/union

🤔

Comment (jdhoffa) [reply src/js/techmix_sector.js:2]: Might be more backwards compatible to use the `d3` implementation? assuming it does the same thing? 

But I'm not sure how to even force it to actually do/ use that, and feels a bit out of scope for this PR

Comment (cjyetman) [reply src/js/techmix_sector.js:2]: Yeah, it must be a new-ish JS function. D3 used to provide stuff like that before they were available in vanilla JS. I guess luckily the syntax/interface/behavior is sufficiently similar.

Comment (jdhoffa) [reply src/js/techmix_sector.js:2]: Well we can monitor it and if things start to seem off, we will know where to look :-)

Comment (cjyetman) [reply src/js/techmix_sector.js:2]: I suspect it is using a vanilla JS Set, and not an old-skool <=D3v4 set/map, but I don't really know the whole evolution of this code anymore, and it's spread out over multiple repos.

https://github.com/RMI-PACTA/pacta-dashboard-svelte/blob/7a24b19a910f9f060adcbe39fc84ae9a510ab991/src/js/techmix_sector.js#L126

**Pull request merged**